src/registration.rs

```diff
@@ -0,0 +1,110 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/// This manages registration of the foxbox with the discovery endpoint.
+/// For now it simply register itselfs every N minutes with the endpoint,
+/// after trying more aggressively at first run.
+
+extern crate get_if_addrs;
+extern crate hyper;
+
+use self::hyper::Client;
+use self::hyper::header::Connection;
+use self::hyper::status::StatusCode;
+use self::get_if_addrs::IfAddr;
+use std::io::Read;
+use std::time::Duration;
+use std::thread;
+
+const DEFAULT_ENDPOINT: &'static str = "http://localhost:4242/register";
+const REGISTRATION_INTERVAL: u64 = 1; // in minutes.
+
+pub struct Registrar;
+
+impl Registrar {
+    pub fn new() -> Registrar {
+        Registrar
+    }
+
+    pub fn start(&self, endpoint: Option<String>) {
+        let url = match endpoint {
+            Some(u) => u,
+            _ => DEFAULT_ENDPOINT.to_owned()
+        };
+
+        info!("Starting registration with {}", url);
+        let ip_addr = self.get_ip_addr();
+        if ip_addr == None {
+            // TODO: retry later, in case we're racing with the network
+            // configuration.
+            return;
+        }
+
+        info!("Got ip address: {}", ip_addr.clone().unwrap());
+        let full_address = format!("{}?ip={}", url, ip_addr.unwrap());
+
+        // Spawn a thread to register every REGISTRATION_INTERVAL minutes.
+        thread::Builder::new().name("Registrar".to_owned())
+                              .spawn(move || {
+            loop {
+                let client = Client::new();
+                let res = client.get(&full_address)
```

Comment:
nit: not to be done in this pr but in the service, perhaps this should be a post.